keras/engine/saving.py

```diff
@@ -333,6 +342,84 @@ def convert_custom_objects(obj):
     return model
 
 
+def _google_storage_transfer(source_filepath, target_filepath, overwrite=True):
+    """Transfers file to/from Google Cloud Storage"""
+    if tf_file_io is None:
+        raise ImportError('Google Storage file transfer requires tensorflow.')
+    if not overwrite:
+        if tf_file_io.file_exists(target_filepath):
+            raise IOError('Object {} already exists.'.format(target_filepath))
+    with tf_file_io.FileIO(source_filepath, mode='r') as source_f:
+        with tf_file_io.FileIO(target_filepath, mode='w') as target_f:
+            target_f.write(source_f.read())
+
+
+def _is_google_storage_location(filepath):
+    """Checks if `filepath` is referencing a google storage bucket"""
+    return isinstance(filepath, string_types) and filepath.startswith('gs://')
+
+
+def parse_save_to_external_resource(save_function):
```

Comment:
I'd suggest more readable/descriptive function names, such as `allow_read_from_gcs` / `allow_write_to_gcs`.

Reply:
Sounds good! I used `external_resources` instead of `gs` to emphasize that we can add support for more things (like S3) but if so it can be renamed later...